web/src/app/services/loi/loi.service.ts

```diff
@@ -134,6 +136,37 @@ export class LocationOfInterestService {
     return null;
   }
 
+  static getLatLngBoundsFromLois(
+    lois: LocationOfInterest[]
+  ): google.maps.LatLngBounds | null {
+    if (!lois.length) return null;
+
+    const data = new google.maps.Data();
+    const bounds = new google.maps.LatLngBounds();
+
+    for (const loi of lois) {
+      if (loi.geometry instanceof Point) {
```

Comment:
This logic isn't specific to LOIs, and there's a bit of duplicated logic in each branch here. Perhaps we can refactor it into individual Geometry classes, with the root call in Coordinate? For ex, MultiPolygon.extendBounds would call Polygon.extendBounds, which would call Coordinate.extendBounds.

Reply:
Yeah, this is a bit cleaner! Doesn't quite work out perfectly, in that polygons never seem to reference points directly for some reason, and I don't feel like it's worth it to refactor at the moment, but still a bit DRYer.